app/coffee/WebsocketController.coffee

```diff
@@ -101,6 +101,29 @@ module.exports = WebsocketController =
 							logger.err {err, project_id, doc_id, fromVersion, line, client_id: client.id}, "error encoding line uri component"
 							return callback(err)
 						escapedLines.push line
+					if options.encodeRanges
+						if ranges.comments
+							escapedComments = []
+							for comment in ranges.comments
+								try
+									comment.op.c = unescape(encodeURIComponent(comment.op.c))
+								catch err
+									logger.err {err, project_id, doc_id, fromVersion, comment, client_id: client.id}, "error encoding comment uri component"
+									return callback(err)
+								escapedComments.push comment
```

Comment:
I don't think we need to collect these in a separate array. We're modifying in place above.